pipeline/pipelines/code_generation_pipeline.py

```diff
@@ -103,10 +116,16 @@ def do_build_flow(self, **kwargs):
                  format_tasks.PythonFormatTask('PythonFormat', inject=kwargs),
                  gapic_tasks.GapicMergeTask('GapicMerge', inject=kwargs),
                  gapic_tasks.GapicPackmanTask('GapicPackman', inject=kwargs))
+        if kwargs['publish_env'] != "none":
+            flow.add(publish_python_gapic_tasks.DevpiIndexTask('DevpiIndex',
+                                                               inject=kwargs),
+                     publish_python_gapic_tasks.DevpiUploadTask('DevpiUpload',
+                                                                inject=kwargs))
         return flow
 
     def validate_kwargs(self, **kwargs):
-        _validate_codegen_kwargs(_VGEN_REQUIRED, **kwargs)
+        _validate_codegen_kwargs(_VGEN_REQUIRED+_PYTHON_PUB_REQUIRED, **kwargs)
```

Comment:
There should be a space around the `+` operator. The linters that we've configured with `tox` -- surprised the linters we have configured with `tox` don't catch this...